src/Common/src/System/Data/Common/AdapterUtil.Drivers.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+
+namespace System.Data.Common
+{
+    internal static partial class ADP
+    {
+
+        internal static Timer CreateGlobalTimer(TimerCallback callback, object state, int dueTime, int period)
+        {
+            // Don't capture the current ExecutionContext and its AsyncLocals onto 
+            // a global timer causing them to live forever
+
+            Timer timer;
```

Comment:
You can just return it from within the try block rather than declaring it here, initializing it in the try, and then returning it after the finally.

Reply:
Changed